engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
@sandeshh the JIRA description talked about "..I was trying to delete different containers ( except the app master ) randomly...."  so I hope that was also tried to reproduce the bug. The stack trace currently pasted in the JIRA description points to allocatedContainer being null (after decoding the synthetic method code).